pkg/manager/member/pump_member_manager.go

```diff
@@ -0,0 +1,475 @@
+					FieldPath: "metadata.name",
+				},
+			},
+		})
+	}
+	containers := []corev1.Container{
+		{
+			Name:            "pump",
+			Image:           spec.Image(),
+			ImagePullPolicy: spec.ImagePullPolicy(),
+			Command: []string{
+				"/bin/sh",
+				"-c",
+				startScript,
+			},
+			Ports: []corev1.ContainerPort{{
+				Name:          "pump",
+				ContainerPort: 8250,
+			}},
+			Resources: util.ResourceRequirement(tc.Spec.Pump.Resources),
+			Env:       envs,
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "data",
+					MountPath: "/data",
+				},
+				{
+					Name:      "config",
+					MountPath: "/etc/pump",
+				},
+			},
+		},
+	}
+
+	// Keep backward compatibility for pump created by helm
+	volumes := []corev1.Volume{
+		{
+			Name: "config",
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: cm.Name,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "pump-config",
+							Path: "pump.toml",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	volumeClaims := []corev1.PersistentVolumeClaim{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "data",
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				StorageClassName: &storageClass,
+				Resources:        *storageRequest,
+			},
+		},
+	}
+
+	podTemplate := corev1.PodTemplateSpec{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: podAnnos,
+			Labels:      pumpLabel,
+		},
+		Spec: corev1.PodSpec{
+			Containers: containers,
+			Volumes:    volumes,
+
+			Affinity:        spec.Affinity(),
+			Tolerations:     spec.Tolerations(),
+			NodeSelector:    spec.NodeSelector(),
+			SchedulerName:   spec.SchedulerName(),
+			SecurityContext: spec.PodSecurityContext(),
+			HostNetwork:     spec.HostNetwork(),
+			DNSPolicy:       spec.DnsPolicy(),
+		},
+	}
+
+	return &appsv1.StatefulSet{
+		ObjectMeta: objMeta,
+		Spec: appsv1.StatefulSetSpec{
+			Selector:    pumpLabel.LabelSelector(),
+			ServiceName: controller.PumpMemberName(tc.Name),
+			Replicas:    &replicas,
+
+			Template:             podTemplate,
+			VolumeClaimTemplates: volumeClaims,
+		},
+	}, nil
+}
+
+func getPumpMeta(tc *v1alpha1.TidbCluster, nameFunc func(string) string) (metav1.ObjectMeta, label.Label) {
+	instanceName := tc.GetLabels()[label.InstanceLabelKey]
+	pumpLabel := label.New().Instance(instanceName).Pump()
+
+	objMeta := metav1.ObjectMeta{
+		Name:            nameFunc(tc.Name),
+		Namespace:       tc.Namespace,
+		Labels:          pumpLabel,
+		OwnerReferences: []metav1.OwnerReference{controller.GetOwnerRef(tc)},
+	}
+	return objMeta, pumpLabel
+}
+
+func getPumpStartScript(tc *v1alpha1.TidbCluster) (string, error) {
+	buff := new(bytes.Buffer)
+	// Keep the logic same as helm chart, but pump has not supported tls yet (no cert mounted)
+	// TODO: support tls
```

Comment:
TLS of pump is not actually supported in tidb-cluster helm chart, nor in controller. We may document this limitation first @AstroProfundis  PTAL